src/project/flowr-analyzer.ts

```diff
@@ -0,0 +1,79 @@
+import type { FlowrConfigOptions } from '../config';
+import { cloneConfig, defaultConfigOptions } from '../config';
+import type { RParseRequests } from '../r-bridge/retriever';
+import { requestFromInput } from '../r-bridge/retriever';
+import type { DEFAULT_DATAFLOW_PIPELINE } from '../core/steps/pipeline/default-pipelines';
+import { createDataflowPipeline, createSlicePipeline } from '../core/steps/pipeline/default-pipelines';
+import type { PipelineOutput } from '../core/steps/pipeline/pipeline';
+import { FlowrAnalyzerBuilder } from './flowr-analyzer-builder';
+import { graphToMermaidUrl } from '../util/mermaid/dfg';
+import type { KnownParser } from '../r-bridge/parser';
+import type { SlicingCriteria } from '../slicing/criterion/parse';
+import type { Queries, SupportedQueryTypes } from '../queries/query';
+import { executeQueries } from '../queries/query';
+import type { DataflowInformation } from '../dataflow/info';
+import type { NormalizedAst } from '../r-bridge/lang-4.x/ast/model/processing/decorate';
+
+export class FlowrAnalyzer {
+	private readonly flowrConfig: FlowrConfigOptions = cloneConfig(defaultConfigOptions);
+	private readonly request:     RParseRequests;
+	private readonly parser:      KnownParser;
+	private ast = undefined as unknown as NormalizedAst;
+	private dataflowInfo = undefined as unknown as DataflowInformation;
+
+	constructor(config: FlowrConfigOptions, parser: KnownParser, request: RParseRequests) {
+		this.flowrConfig = config;
+		this.request = request;
+		this.parser = parser;
+	}
+
+	public async dataflow() : Promise<PipelineOutput<typeof DEFAULT_DATAFLOW_PIPELINE>> {
```

Comment:
Using default data flow pipeline here is... Wrong in general we should select the type based on the engine :D in general we maybe want to embed a cache?

Reply:
> in general we maybe want to embed a cache?

Yes, I think that would make sense. We need to have the dataflow analysis results for the query execution anyways. In theory, the slicing pipeline could re-use the dataflow step results, but I'm not sure whether we should enable this right now